tests/utils/test_file.py

```diff
@@ -57,12 +57,24 @@ def test_fail_create_dir():
 
 def test_walk_tree():
     list_of_files = walk_tree(here, ".py")
+    print(f"Detected {len(list_of_files)} .py files found in directory: {here}")
     assert len(list_of_files) > 0
 
 
-@pytest.mark.xfail(
-    reason="This test is expected to fail since we passed invalid path for directory traversal",
-    raises=BuildTestError,
-)
-def test_walk_tree_invalid_dir():
-    walk_tree("/xyz", ".py")
+def test_walk_tree_no_files(tmp_path):
+    list_of_files = walk_tree(tmp_path, ".py")
+    print(f"Detected {len(list_of_files)} .py files found in directory: {here}")
```

Comment:
Is this really testing walking a tree, if there will never be a tree in this location?

Reply:
Ok there is a typo in the print statement, i fixed locally but yes it is doing a directory traversal in tmp_path expecting to find no .py files. This is different than the other test ``test_walk_tree_invalid_dir`` which is testing if invalid directory is specified. 
```
(buildtest-framework) ssi29@ag-mxg-hulk090> pytest -vrP tests/utils/test_file.py
============================================================================================= test session starts =============================================================================================
platform linux -- Python 3.7.4, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /u/users/ssi29/.local/share/virtualenvs/buildtest-framework-LDt8wyRf/bin/python3.7
cachedir: .pytest_cache
rootdir: /mxg-hpc/users/ssi29/buildtest-framework
collected 8 items

tests/utils/test_file.py::test_checking_directory PASSED                                                                                                                                                [ 12%]
tests/utils/test_file.py::test_checking_file XPASS                                                                                                                                                      [ 25%]
tests/utils/test_file.py::test_directory_expansion PASSED                                                                                                                                               [ 37%]
tests/utils/test_file.py::test_create_dir PASSED                                                                                                                                                        [ 50%]
tests/utils/test_file.py::test_fail_create_dir XFAIL                                                                                                                                                    [ 62%]
tests/utils/test_file.py::test_walk_tree PASSED                                                                                                                                                         [ 75%]
tests/utils/test_file.py::test_walk_tree_no_files PASSED                                                                                                                                                [ 87%]
tests/utils/test_file.py::test_walk_tree_invalid_dir PASSED                                                                                                                                             [100%]

=================================================================================================== PASSES ====================================================================================================
_______________________________________________________________________________________________ test_walk_tree ________________________________________________________________________________________________
-------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------
Detected 3 .py files found in directory: /mxg-hpc/users/ssi29/buildtest-framework/tests/utils
___________________________________________________________________________________________ test_walk_tree_no_files ___________________________________________________________________________________________
-------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------
Detected 0 .py files found in directory: /scratch/pytest-of-ssi29/pytest-1/test_walk_tree_no_files0
_________________________________________________________________________________________ test_walk_tree_invalid_dir __________________________________________________________________________________________
-------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------
Removing directory: /scratch/pytest-of-ssi29/pytest-1/test_walk_tree_invalid_dir0 first before doing directory traversal on invalid directory
Returned following files: [] with .py extension for path: /scratch/pytest-of-ssi29/pytest-1/test_walk_tree_invalid_dir0
=================================================================================== 6 passed, 1 xfailed, 1 xpassed in 0.93s ===================================================================================
```